src/dodal/beamlines/i22.py

```diff
@@ -42,339 +46,250 @@
-    return device_instantiation(
-        FocusingMirror,
-        "hfm",
-        "-OP-KBM-01:HFM:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory()
+def hfm() -> FocusingMirror:
+    return FocusingMirror(name="hfm", prefix="-OP-KBM-01:HFM:")
+
+
+@device_factory()
+def dcm() -> DoubleCrystalMonochromator:
+    prefix = BeamlinePrefix(BL).beamline_prefix
+    silicon_111 = CrystalMetadata(
+        usage="Bragg",
+        type="silicon",
+        reflection=(1, 1, 1),
+        d_spacing=(3.13475, "nm"),
     )
 
+    silicon_311 = CrystalMetadata(
+        usage="Bragg",
+        type="silicon",
+        reflection=(3, 1, 1),
+        # todo update
+        d_spacing=(3.13475, "nm"),
+    )
 
-def dcm(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> DoubleCrystalMonochromator:
-    return device_instantiation(
-        DoubleCrystalMonochromator,
-        "dcm",
-        "",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-        bl_prefix=False,
-        motion_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
-        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
-        crystal_1_metadata=CrystalMetadata(
-            usage="Bragg",
-            type="silicon",
-            reflection=(1, 1, 1),
-            d_spacing=(3.13475, "nm"),
-        ),
-        crystal_2_metadata=CrystalMetadata(
-            usage="Bragg",
-            type="silicon",
-            reflection=(1, 1, 1),
-            d_spacing=(3.13475, "nm"),
-        ),
+    return DoubleCrystalMonochromator(
+        motion_prefix=f"{prefix}-MO-DCM-01:",
+        temperature_prefix=f"{prefix}-DI-DCM-01:",
+        crystal_1_metadata=silicon_111,
+        crystal_2_metadata=silicon_311,
     )
 
 
-def undulator(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> Undulator:
-    return device_instantiation(
-        Undulator,
-        "undulator",
-        f"{BeamlinePrefix(BL).insertion_prefix}-MO-SERVC-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-        bl_prefix=False,
+@device_factory()
+def undulator() -> Undulator:
+    return Undulator(
+        prefix=f"{BeamlinePrefix(BL).insertion_prefix}-MO-SERVC-01:",
         poles=80,
         length=2.0,
         id_gap_lookup_table_path="/dls_sw/i22/software/daq_configuration/lookup/BeamLine_Undulator_toGap.txt",
     )
 
 
-def slits_1(
-    wait_for_connection: bool = True,
-    fake_with_ophyd_sim: bool = False,
-) -> Slits:
-    return numbered_slits(
-        1,
-        wait_for_connection,
-        fake_with_ophyd_sim,
+# slits section, s1 is on the frontend, s6 is closest to the sample
+@device_factory()
+def slits_1() -> Slits:
+    slit_number = 1
+    return device_instantiation(
+        Slits,
+        f"slits_{slit_number}",
+        f"-AL-SLITS-{slit_number:02}:",
+        wait_for_connection=False,
+        fake_with_ophyd_sim=False,
     )
```

Comment:
Must: The issue that you were seeing with the slits before was because `numbered_slits` uses `device_instantiation`, which is the old way of making devices, rather than the new way which is just call the constructor like you normally would. This change still has that issue except that you have changed the `device_instantiation` to default to `fake_with_ophyd_sim=False`, which will now break in prod. You can keep the old structure, and make it work properly, by doing:
```python
def numbered_slits(slit_number):
    return Slits(f"-AL-SLITS-{slit_number:02}:", f"slits_{slit_number}")

@device_factory()
def slits_1() -> Slits:
    return slits_helper(1)
```

Reply:
ah my bad I was rushing this of course I need this

`    return Slits(f"-AL-SLITS-{slit_number:02}:", f"slits_{slit_number}")`

I don't like the helper though

Reply:
Why don't you like the helper? I think given how much shorter this is now I wouldn't be adverse to dropping the helper.